dockerspawner/dockerspawner.py

```diff
@@ -373,6 +373,20 @@ def get_container(self):
                 raise
         return container
 
+    def upgrade_kwags(self, data):
+        import sys
+        if data is not None:
+            for k, v in data.items():
+                w = v.format(username=self.user.name, escapedusername=self.escaped_name)
```

Comment:
We use some template formatting in a few places. It would probably be better to make the existing volume_formatting use a standard template  and namespace to consolidate these things, so that every templated string gets access to the same information.

Reply:
I agree. 
I just couldn't spend a lot of time on it so i copied your code into mine.
A single point for applying it would be more desirable indeed